### PR TITLE
Do not abort unsupported property type

### PR DESCRIPTION
**Pull request by blaxter**
Mostly refactorization. It only removes some `abort()` and now we will return error on those scenarios.

### CODE REVIEW

File: mapiproxy/libmapiserver/libmapiserver_oxcprpt.c [hunk 1]
```diff
@@ -405,19 +405,16 @@ _PUBLIC_ uint16_t libmapiserver_RopDeletePropertiesNoReplicate_size(struct EcDoR
...
-_PUBLIC_ int libmapiserver_push_property(TALLOC_CTX *mem_ctx,
-					 uint32_t property, 
-					 const void *value, 
-					 DATA_BLOB *blob,
-					 uint8_t layout, 
-					 uint8_t flagged,
+_PUBLIC_ int libmapiserver_push_property(TALLOC_CTX *mem_ctx, uint32_t property,
+					 const void *value, DATA_BLOB *blob,
+					 uint8_t layout, uint8_t flagged,
 					 uint8_t untyped)
```
Comment (sixstone-qq) [inline mapiproxy/libmapiserver/libmapiserver_oxcprpt.c]: New parameter is not documented.

Didn't we agree to set the documention on header files?

File: mapiproxy/servers/default/emsmdb/oxctabl.c [hunk 2]
```diff
@@ -862,36 +861,22 @@ _PUBLIC_ enum MAPISTATUS EcDoRpc_RopFindRow(TALLOC_CTX *mem_ctx,
...
-					}
-                                
-					libmapiserver_push_property(mem_ctx,
-								    property, data, &row,
-								    flagged?PT_ERROR:0, flagged, 0);
-				}
+
+				libmapiserver_push_properties(mem_ctx, table->prop_count,
+					table->properties, data_pointers, retvals,
+					&row, flagged?PT_ERROR:0, flagged, 0);
```
Comment (sixstone-qq) [inline mapiproxy/servers/default/emsmdb/oxctabl.c]: Missing spaces?

File: mapiproxy/servers/default/emsmdb/oxctabl.c [hunk 3]
```diff
@@ -932,35 +917,23 @@ _PUBLIC_ enum MAPISTATUS EcDoRpc_RopFindRow(TALLOC_CTX *mem_ctx,
...
```
Comment (sixstone-qq) [inline mapiproxy/servers/default/emsmdb/oxctabl.c]: Missing?

File: mapiproxy/libmapiserver/libmapiserver.h [hunk 4]
```diff
@@ -538,6 +538,8 @@ uint16_t libmapiserver_RopGetPropertyIdsFromNames_size(struct EcDoRpc_MAPI_REPL
 uint16_t libmapiserver_RopDeletePropertiesNoReplicate_size(struct EcDoRpc_MAPI_REPL *);
 uint16_t libmapiserver_RopCopyTo_size(struct EcDoRpc_MAPI_REPL *);
 int libmapiserver_push_property(TALLOC_CTX *, uint32_t, const void *, DATA_BLOB *, uint8_t, uint8_t, uint8_t);
+void libmapiserver_push_properties(TALLOC_CTX *, size_t, enum MAPITAGS *, void **, enum MAPISTATUS *, DATA_BLOB *, uint8_t, uint8_t, bool);
+void libmapiserver_push_properties_with_untyped(TALLOC_CTX *, size_t, enum MAPITAGS *, void **, enum MAPISTATUS *, DATA_BLOB *, uint8_t, uint8_t, bool *);
```
Comment (sixstone-qq) [inline mapiproxy/libmapiserver/libmapiserver.h]: Why don't you include the arg name variables and the doc here?

Comment (blaxter) [inline mapiproxy/libmapiserver/libmapiserver.h]: Because I suck

File: mapiproxy/servers/default/emsmdb/oxctabl.c (hunk 3 above)
Comment (blaxter) [inline mapiproxy/servers/default/emsmdb/oxctabl.c]: Indeed

File: mapiproxy/servers/default/emsmdb/oxctabl.c (hunk 2 above)
Comment (blaxter) [inline mapiproxy/servers/default/emsmdb/oxctabl.c]: Indeed

File: mapiproxy/libmapiserver/libmapiserver_oxcprpt.c (hunk 1 above)
Comment (blaxter) [inline mapiproxy/libmapiserver/libmapiserver_oxcprpt.c]: I'll do my best, but first I have to understand it. Get back to you soon

Comment (blaxter) [inline mapiproxy/libmapiserver/libmapiserver_oxcprpt.c]: Basically this means that the property (to be pushed) was originally untyped (we didn't know the prop type but we did know the prop id) and the prop type has been obtained checking the known properties, in that case it seems that the prop type is pushed as well on the `DATA_BLOB`. Interesting

TL;DR: documented

**Comment by icandigitbaby** [comment]
:+1:

**Comment by sixstone-qq** [comment]
:+1:
